### PR TITLE
Wallet adapter

### PR DESCRIPTION
**Pull request by MasterSlave**

### CODE REVIEW

File: Craftgate/Adapter/WalletAdapter.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+        public WalletListResponse SearchWallets(SearchWalletRequest searchWalletRequest)
+        {
+            var queryParam = RequestQueryParamsBuilder.BuildQueryParam(searchWalletRequest);
+            var path = "/wallet/v1/wallets" + queryParam;
+            return RestClient.Get<WalletListResponse>(RequestOptions.BaseUrl + path,
+                CreateHeaders(path, RequestOptions));
+        }
+
+        public WalletTransactionListResponse SearchWalletsTransactions(long walletId,
```
Comment (sotuzun) [inline Craftgate/Adapter/WalletAdapter.cs:22]: SearchWalletTransactions

File: Craftgate/Adapter/WalletAdapter.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+namespace Craftgate.Adapter
+{
+    public class WalletAdapter : BaseAdapter
+    {
+        public WalletAdapter(RequestOptions requestOptions) : base(requestOptions)
+        {
+        }
+
+        public WalletListResponse SearchWallets(SearchWalletRequest searchWalletRequest)
```
Comment (sotuzun) [inline Craftgate/Adapter/WalletAdapter.cs:14]: SearchWalletsRequest

File: Craftgate/Adapter/WalletAdapter.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        {
+            var queryParam = RequestQueryParamsBuilder.BuildQueryParam(searchWalletRequest);
+            var path = "/wallet/v1/wallets" + queryParam;
+            return RestClient.Get<WalletListResponse>(RequestOptions.BaseUrl + path,
+                CreateHeaders(path, RequestOptions));
+        }
+
+        public WalletTransactionListResponse SearchWalletsTransactions(long walletId,
+            SearchWalletTransactionRequest searchWalletTransactionRequest)
```
Comment (sotuzun) [inline Craftgate/Adapter/WalletAdapter.cs:23]: SearchWalletTransactionsRequest

File: Craftgate/Adapter/PaymentAdapter.cs [hunk 4]
```diff
@@ -27,36 +27,52 @@ public PaymentResponse RetrievePayment(long id)
...
+        public DepositPaymentRefundResponse RefundDepositPayment(long paymentId,
+            RefundDepositPaymentRequest refundDepositPaymentRequest)
+        {
+            var path = "/payment/v1/deposits/" + paymentId + "/refunds";
+            return RestClient.Post<DepositPaymentRefundResponse>(RequestOptions.BaseUrl + path,
+                CreateHeaders(refundDepositPaymentRequest, path, RequestOptions),
+                refundDepositPaymentRequest);
+        }
+
+        public InitThreeDSPaymentResponse Init3DSDepositPayment(InitThreeDSPaymentRequest initThreeDSPaymentRequest)
```
Comment (sotuzun) [inline Craftgate/Adapter/PaymentAdapter.cs:63]: CreateDepositPaymentRequest

**Review by sotuzun — DISMISSED** [review]

File: Craftgate/Adapter/PaymentAdapter.cs (hunk 4 above)
Comment (MasterSlave) [reply Craftgate/Adapter/PaymentAdapter.cs]: done

File: Craftgate/Adapter/WalletAdapter.cs (hunk 3 above)
Comment (MasterSlave) [reply Craftgate/Adapter/WalletAdapter.cs]: done

File: Craftgate/Adapter/WalletAdapter.cs (hunk 2 above)
Comment (MasterSlave) [reply Craftgate/Adapter/WalletAdapter.cs]: done

File: Craftgate/Adapter/WalletAdapter.cs (hunk 1 above)
Comment (MasterSlave) [reply Craftgate/Adapter/WalletAdapter.cs]: done

**Review by sotuzun — APPROVED** [review]

**Pull request merged**